crossdock/src/helpers.js

```diff
@@ -20,10 +20,196 @@
 // THE SOFTWARE.
 
 import * as constants from './constants.js';
+import fs from 'fs';
+import path from 'path';
+import dns from 'dns';
+import opentracing from 'opentracing';
+import request from 'request';
+import RSVP from 'rsvp';
+import Span from '../../src/span.js';
+import SpanContext from '../../src/span_context.js';
+import TChannel from 'tchannel/channel';
+import TChannelThrift from 'tchannel/as/thrift';
+import TChannelBridge from '../../src/tchannel_bridge';
 import Utils from '../../src/util.js';
 
 export default class Helpers {
-    static observeSpan(context: any): ObservedSpan {
+    _tracer: Tracer;
+    _bridge: TChannelBridge;
+    _thriftChannel: Channel;
+
+    constructor(tracer: Tracer) {
+        this._tracer = tracer;
+
+        var channel = TChannel().makeSubChannel({
+            serviceName: 'node',
+            peers: [Utils.myIp() + ':8082']
+        });
+
+        let crossdockSpec = fs.readFileSync(path.join(__dirname, '..', '..', 'src', 'jaeger-idl', 'thrift', 'crossdock', 'tracetest.thrift'), 'utf8');
+        this._thriftChannel = TChannelThrift({
+            channel: channel,
+            source: crossdockSpec
+        });
+
+        let bridge = new TChannelBridge(this._tracer);
+        this._tracedChannel= bridge.tracedChannel(this._thriftChannel);
+    }
+
+    handleRequest(isStartRequest: boolean, traceRequest: any, spanOrContext: any, operationName: string): void {
+        let serverSpan;
+        if (spanOrContext instanceof Span) {
```

Comment:
I don't see why tchannel handler passes `context.openTracingSpan.context()` instead of just `context.openTracingSpan`. If it did the latter, you wouldn't have needed these type checks